client/dogs/SearchPage.js

```diff
@@ -85,35 +74,31 @@ export default class IndexPage extends React.Component {
   };
 
   handleSubmit = (ev: SyntheticInputEvent) => {
-    const { query } = this.state;
+    const query = this.state;
```

Comment:
I'd remove this whole method; doing a `toString` on the `state` object won't actually build up a query parameter string. This isn't called now anyway since you're doing the normal form GET method below.